packages/Main/src/Source/VpcSource.js

```diff
@@ -0,0 +1,131 @@
+
+        this.colorDepth = config.colorDepth;
+
+        this.spacing = Infinity;
+
+        this.whenReady = Fetcher.json(this.url, this.networkOptions)
+            .then((metadata) => {
+                this.metadata = metadata;
+                const boundsConformings = metadata.features.map(f => f.properties['proj:bbox']);
+
+                this.boundsConforming = [
+                    Math.min(...boundsConformings.map(b => b[0])),
+                    Math.min(...boundsConformings.map(b => b[1])),
+                    Math.min(...boundsConformings.map(b => b[2])),
+                    Math.max(...boundsConformings.map(b => b[3])),
+                    Math.max(...boundsConformings.map(b => b[4])),
+                    Math.max(...boundsConformings.map(b => b[5])),
+                ];
+
+                this.minElevation = this.boundsConforming[2];
+                this.maxElevation = this.boundsConforming[5];
+
+                // Crs
+                const projsWkt2 = [...new Set(metadata.features.map(f => f.properties['proj:wkt2']))];
+                if (projsWkt2.length !== 1) {
+                    console.warn('Only 1 crs is currently supported for 1 vpc. The extra crs will not be considered');
+                }
+                proj4.defs('unknown', projsWkt2[0]);
+                let projCS;
+                if (proj4.defs('unknown').type === 'COMPD_CS') {
+                    console.warn('CopcSource: compound coordinate system is not yet supported.');
+                    projCS = proj4.defs('unknown').PROJCS;
+                } else {
+                    projCS = proj4.defs('unknown');
+                }
+                this.crs = projCS.title || projCS.name || 'EPSG:4326';
+                if (!(this.crs in proj4.defs)) {
+                    proj4.defs(this.crs, projCS);
+                }
+
+                // Creation of 1 mockSource for each item in the stack (that will be replace when instanciate)
+                this._promises = [];
+                this.urls = metadata.features.map(f => f.assets.data.href);
+                this.urls.forEach((url, i) => {
+                    const whenReady = new Promise((re, rj) => {
+                        // waiting for this.instanciate(source);
+                        this._promises.push({ resolve: re, reject: rj });
+                    }).then((res) => {
+                        this.sources[i] = res;
+                        return res;
+                    }).catch((err) => {
+                        console.warn(err);
+                        this.handlingError(err);
+                    });
+
+                    const mockSource = {
+                        url,
+                        boundsConforming: boundsConformings[i],
+                        whenReady,
+                        sId: i,
+                    };
+                    this.sources.push(mockSource);
+                });
+
+                return this.sources;
```

Comment:
I think this then should be split into several simpler methods, it's too complex to understand like this in my opinion. You could for example have one method for the bounds, one for the crs and one for the sources manipulations.